### PR TITLE
LPS-95863 Format module's API definition

### PR DESCRIPTION
**Pull request by jbalsas**
Follow up of https://github.com/wincent/liferay-portal/pull/54
- Consolidates all `store` functions in `session` instead
- Gets rid of `getAll` which clearly shows unwanted behaviours already in the old implementation
- Reverted the old util to its old implementation. Not worth the risk

### CODE REVIEW

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/index.es.js [hunk 1]
```diff
@@ -12,42 +12,91 @@
...
+	default as getStoreValues
+} from './liferay/util/store/get_store_values.es';
+
+export {
+	default as setStoreValue
+} from './liferay/util/store/set_store_value.es';
+
+export {
+	default as setStoreValues
+} from './liferay/util/store/set_store_values.es';
```
Comment (wincent) [inline modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/index.es.js]: These refer to files which should be deleted now, right? I think we should delete those files and these exports.

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/global.es.js [hunk 2]
```diff
@@ -96,6 +103,12 @@ Liferay.Util.openToast = (...args) => {
 	);
 };
 
+Liferay.Util.Session = {
+	get: getSessionValue,
+	set: setSessionValue,
+	setAll: setSessionValues
```
Comment (wincent) [inline modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/global.es.js]: Seems we should drop `setAll` and `setSessionValues` too; no point in having an asymmetric API.

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/session.es.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+ * @review
+ */
+export function setSessionValues(data) {
+	const formData = getSessionClickFormData('setAll');
+
+	return fetch(getSessionClickURL().toString(), {
+		body: objectToFormData(data, formData),
+		method: 'POST'
+	});
+}
```
Comment (wincent) [inline modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/session.es.js]: Delete this function.

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/session.es.js [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+	const formData = getSessionClickFormData('get');
+
+	formData.append('key', key);
+
+	return fetch(getSessionClickURL().toString(), {
+		body: formData,
+		method: 'POST'
+	})
+		.then(response => response.text())
+		.then(responseText => processSessionResponse(responseText));
```
Comment (wincent) [inline modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/session.es.js]: You use `processSessionResponse` in exactly one place; you could just inline it.

**Comment by jbalsas** [comment]
Sent to Brian -> https://github.com/brianchandotcom/liferay-portal/pull/77376 

Mind taking a final look?

File: modules/apps/frontend-js/frontend-js-web/test/liferay/util/session.es.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+	setSessionValue
+} from '../../../src/main/resources/META-INF/resources/liferay/util/session.es';
+
+describe('Session API', () => {
+	beforeEach(() => {
+		fetch.mockResponse('');
+	});
+
+	describe('getSessionValue', () => {
+		it('GETs the session_click endpoint', () => {
```
Comment (wincent) [inline modules/apps/frontend-js/frontend-js-web/test/liferay/util/session.es.js]: Only thing is, now it doesn't `GET`, it `POST`s 😂.

**Pull request closed without merging**